### ISSUE TITLE
"Add This Product" button becomes active only after a few secconds on slow connections.

### ISSUE THREAD

**Issue by muccimoz** [mozilla/price-wise#242]
I have verified this issue using the latest version of the Price Wise (8.0.0-signed) version and the issue is still reproducible on https://www.walmart.com/, using a slow Internet connection. The "Add This Product" button is briefly active after the page is fully loaded and after that is no longer active.

However, on https://www.homedepot.com/, after the page is fully loaded, the "Add This Product" button becomes active only after a few secconds. 

In order to test this I have simulated a slow Internet connection using the "Network Link Conditioner" app.,on Mac, and also with the functionality from RDM mode.
Attached is a screen recording of the issue:
 
![slowinternet](https://user-images.githubusercontent.com/40061643/47863006-0b40df80-ddff-11e8-80d5-ff4aeb88d261.gif)

_Originally posted by @Softvision-RemusDranca in https://github.com/mozilla/price-wise/issues/192#issuecomment-435089865_

**Comment by Osmose** [comment]
@Softvision-RemusDranca I've made a version of the add-on with the original workaround we had: https://usercontent.irccloud-cdn.com/file/LCrGGoJF/price_wise-8.0.1-signed.xpi

Using that add-on on release, I can't replicate the bug at all, on slow or fast connections. Can you confirm whether you can replicate the issue with it?

**Comment by Softvision-RemusDranca** [comment]
I have retested this issue with a slow connection using the provided Price Wise (8.0.1-signed) version, and the issue is no longer reproducible. 
However, I have observed that the green badge flickers while the page is loading. Is more visible on slow connections, but this can also be observed on good connection. Should I log a new issue for this behavior, for tracking purpose? 
Tested on Windows 10 x64, Mac 10.13 and Ubuntu 18.04 x64.
Here is a screen recording:
![flicker](https://user-images.githubusercontent.com/40061643/48006065-a39bd480-e11d-11e8-9f69-3ef295ebe719.gif)
